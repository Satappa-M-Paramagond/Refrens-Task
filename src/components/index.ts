export { FormatText } from './FormatText';
export { Loader } from './Loader';
export { PrimaryButton } from './PrimaryButton';
export { UserProfileCard } from './UserProfileCard';
