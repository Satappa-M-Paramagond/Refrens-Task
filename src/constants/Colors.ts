export const Colors = {
    black: '#000000',
    blue: '#233f78',
    cyan: 'cyan',
    darkCyan: '#004053',
    darkGrey: 'darkgrey',
    grey: 'grey',
    lightCyan: '#87CBCF',
    lighterCyan: '#364C4D',
    lightGrey: 'lightgrey',
    orange: '#f37421',
    transparent: 'transparent',
    white: '#ffffff',
    whiteSmoke: 'whitesmoke'
};
