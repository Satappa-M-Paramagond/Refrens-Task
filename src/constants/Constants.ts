export const GetIDFromURL = (url: string) => {
    let id = 0;
    if (url) {
        id = url.split('/').pop();
    }
    return id;
};

export const TestIDs = {
    primaryButton: 'primaryButton',
    profileCard: 'profileCard'
};
