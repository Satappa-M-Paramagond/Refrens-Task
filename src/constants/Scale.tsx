import { Dimensions } from 'react-native';
const { width } = Dimensions.get('window');

// Sample device width to calculate scaling
const guidelineBaseWidth = 375;

export const Scale = (size: number) =>
    size + ((width / guidelineBaseWidth) * size - size) * 0.5;
