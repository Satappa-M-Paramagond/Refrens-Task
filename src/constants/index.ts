export { Colors } from './Colors';
export { GetIDFromURL, TestIDs } from './Constants';
export { Scale } from './Scale';
export { Screens } from './Screens';
export { Strings } from './Strings';
