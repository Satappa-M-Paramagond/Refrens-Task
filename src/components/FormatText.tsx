import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Colors, Scale } from '../constants';

type Props = {
    title: string;
    text: string;
};

export const FormatText: React.FC<Props> = props => {
    const { title = '', text = '' } = props;

    if (text && !!text) {
        return <Text style={styles.itemText}>{`${title} : ${text}`}</Text>;
    } else {
        return null;
    }
};

const styles = StyleSheet.create({
    itemText: {
        fontSize: Scale(12),
        color: Colors.black
    }
});
