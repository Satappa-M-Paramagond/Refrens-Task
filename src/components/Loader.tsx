import React from 'react';
import { StyleSheet, View, ActivityIndicator, ViewStyle } from 'react-native';
import { Colors } from '../constants';

enum Sizes {
    SMALL = 'small',
    LARGE = 'large'
}
type Props = {
    containerStyle?: ViewStyle;
    size?: Sizes;
};

export const Loader: React.FC<Props> = props => {
    const { containerStyle = {}, size = 'large' } = props;

    return (
        <View style={[styles.container, containerStyle]}>
            <ActivityIndicator color={Colors.darkGrey} size={size} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
});
