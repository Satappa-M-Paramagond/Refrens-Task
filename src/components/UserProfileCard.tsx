import React, { useCallback } from 'react';
import { NavigationAction } from '@react-navigation/native';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Colors, Scale, Screens, Strings, TestIDs } from '../constants';
import { FormatText } from './FormatText';

type Location = {
    name: string;
    url: string;
};

type UserInfo = {
    name: string;
    gender: string;
    species: string;
    status: string;
    origin?: Location;
    location?: Location;
    image: string;
};

type Props = {
    userInfo: UserInfo;
    navigation?: any;
};

export const UserProfileCard: React.FC<Props> = props => {
    const { userInfo, navigation } = props;

    const handleProfilePress = useCallback(() => {
        navigation.navigate(Screens.PROFILE, {
            characterDetails: userInfo
        });
    }, [navigation, userInfo]);

    return (
        <View style={styles.profileCard}>
            <TouchableOpacity
                testID={TestIDs.profileCard}
                accessibilityLabel={TestIDs.profileCard}
                onPress={handleProfilePress}>
                <View style={styles.userContainer}>
                    <Image
                        source={{ uri: userInfo?.image }}
                        style={styles.profilePic}
                    />
                    <View style={styles.userDetailsContainer}>
                        <Text style={styles.userName}>{userInfo.name}</Text>
                        <FormatText
                            title={Strings.gender}
                            text={userInfo.gender}
                        />
                        <FormatText
                            title={Strings.species}
                            text={userInfo.species}
                        />
                        <FormatText
                            title={Strings.status}
                            text={userInfo.status}
                        />
                    </View>
                </View>
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    profileCard: {
        flex: 1,
        marginHorizontal: Scale(20),
        marginVertical: Scale(8),
        backgroundColor: Colors.white,
        borderRadius: Scale(10),
        padding: Scale(12),
        elevation: 4,
        shadowOffset: {
            width: 3,
            height: 3
        },
        shadowColor: Colors.lightGrey,
        shadowOpacity: 1
    },
    userContainer: {
        flexDirection: 'row'
    },
    profilePic: {
        height: Scale(66),
        width: Scale(66),
        borderRadius: Scale(10),
        alignSelf: 'center'
    },
    userDetailsContainer: {
        paddingHorizontal: Scale(10),
        flexShrink: 1
    },
    userName: {
        marginBottom: Scale(2),
        color: Colors.blue,
        fontSize: Scale(16),
        fontWeight: '600',
        textAlign: 'left'
    }
});
