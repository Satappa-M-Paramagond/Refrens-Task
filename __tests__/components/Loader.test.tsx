jest.useFakeTimers();
import React from 'react';
import renderer from 'react-test-renderer';
import { Loader } from '../../src/components';
import { Colors } from '../../src/constants';

describe('Loader => Snapshots => ', () => {
    test('should render properly without props', () => {
        const tree = renderer.create(<Loader />).toJSON();
        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });

    test('should render properly with props', () => {
        const tree = renderer
            .create(
                <Loader
                    containerStyle={{ backgroundColor: Colors.lightGrey }}
                    size={'large'}
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });
});
