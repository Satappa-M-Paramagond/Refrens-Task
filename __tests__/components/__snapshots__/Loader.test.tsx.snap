// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader => Snapshots =>  should render properly with props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "lightgrey",
      },
    ]
  }
>
  <ActivityIndicator
    color="darkgrey"
    size="large"
  />
</View>
`;

exports[`Loader => Snapshots =>  should render properly without props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ActivityIndicator
    color="darkgrey"
    size="large"
  />
</View>
`;
